amundsen_application/static/js/ducks/search/types.ts

```diff
@@ -42,6 +42,7 @@ export interface SearchAllRequest {
   options: SearchAllOptions;
   term: string;
   type: SearchAll.ACTION;
+  payload?: SearchReducerState;
```

Comment:
I don't think the `SearchAllRequest` ever has a `payload`. I'm guessing you added this to suppress an error on line 61 of `search/reducer.ts`. This is caused because not every action in the reducer function is guaranteed to have a `payload`.

There are some alternatives. You could:

1. Move `const newState = action.payload` from `reducer.ts` line 61 into the sections on line 70, 76. Or you can just reference `action.payload` directly.

2. Require `SearchAllRequest` to have a payload and move `term` and `type` into payload. This goes against the established pattern we have where none of the Requests have a payload, but I wouldn't be opposed to following this pattern for all redux actions. @ttannis would also need to get onboard here. There are a few patterns that I have stronger opinions about now that I am more familiar with redux/saga.

Reply:
Spoke offline. Fine with this change and I'll pick up the task to clean up what we're doing here alongside unit test additions.